src/server/publicApi.js

```diff
@@ -60,14 +56,37 @@ const runPublicApi = (gameServer) => {
     };
   });
 
-  router.get('/model/:id', async (ctx) => {
-    const matchID = ctx.params.id;
+  //authorise
+  router.use('/:matchID/', async (ctx, next) => {
+    const user = auth(ctx);
```

Comment:
Maybe call this `credentials`? My first thought was: "why is the username compared to the password?"